metasync.js

```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
   if (len > 0) next();
   else done();
 };
+
+
+// Asynchrous filter
+// filter :: (a -> Boolean) -> [a] -> ([a] -> Void)
+metasync.filter = function(predicate, array, done) {
+  var result = [];
+  var index = 0;
+  
+  function doFilter() {
+    if (index === array.length) {
+      done(result);
+    } else {
+      if (predicate(array[index])) {
```

Comment:
Because those functions contains asynchronous operations, I/O (files, network, database, etc.), timers, events and so on. That's asynchronous programming, functions should return only using callbacks.